internal/service/encryptionatrestprivateendpoint/pural_data_source_schema.go

```diff
@@ -0,0 +1,114 @@
+package encryptionatrestprivateendpoint
+
+import (
+	"context"
+
+	"go.mongodb.org/atlas-sdk/v20240805001/admin"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
+)
+
+func PluralDataSourceSchema(ctx context.Context) schema.Schema {
+	return schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"cloud_provider": schema.StringAttribute{
+				Required:            true,
+				Description:         "Human-readable label that identifies the cloud provider for the private endpoints to return.",
+				MarkdownDescription: "Human-readable label that identifies the cloud provider for the private endpoints to return.",
+			},
+			"project_id": schema.StringAttribute{
+				Required:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies your project. Use the [/groups](#tag/Projects/operation/listProjects) endpoint to retrieve all projects to which the authenticated user has access.\n\n**NOTE**: Groups and projects are synonymous terms. Your group id is the same as your project id. For existing groups, your group/project id remains the same. The resource and corresponding endpoints use the term groups.",
+			},
+			"results": schema.ListNestedAttribute{
+				NestedObject: schema.NestedAttributeObject{
+					Attributes: map[string]schema.Attribute{
+						"cloud_provider": schema.StringAttribute{
```

Comment:
can we reuse the data_source schema similar to [stream-connection DS](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/internal/service/streamconnection/data_source_stream_connection.go#L35) or do we prefer to keep them independent?

Reply:
There's minor difference like presence of project_id in the singular DS model so they aren't exactly the same, I'd prefer to keep the autogenerated schema code close to the API implementation.